src/python_testing/TC_AVSM_2_10.py

```diff
@@ -0,0 +1,215 @@
+        logger.info(f"Rx'd FeatureMap: {aFeatureMap}")
+        snpSupport = (aFeatureMap & cluster.Bitmaps.Feature.kSnapshot) > 0
+        asserts.assert_true(snpSupport, "Snapshot Feature is not supported.")
+
+        # Check if snapshot stream has already been allocated
+        aAllocatedSnapshotStreams = await self.read_single_attribute_check_success(
+            endpoint=endpoint, cluster=cluster, attribute=attr.AllocatedSnapshotStreams
+        )
+        logger.info(f"Rx'd AllocatedSnapshotStreams: {aAllocatedSnapshotStreams}")
+        if len(aAllocatedSnapshotStreams) > 0:
+            return
+
+        # Allocate one for the test steps based on SnapshotCapabilities
+        aSnapshotCapabilities = await self.read_single_attribute_check_success(
+            endpoint=endpoint, cluster=cluster, attribute=attr.SnapshotCapabilities
+        )
+        logger.info(f"Rx'd SnapshotCapabilities: {aSnapshotCapabilities}")
+
+        asserts.assert_greater(len(aSnapshotCapabilities), 0, "SnapshotCapabilities list is empty")
+        try:
+            snpStreamAllocateCmd = commands.SnapshotStreamAllocate(
+                imageCodec=aSnapshotCapabilities[0].imageCodec,
+                maxFrameRate=aSnapshotCapabilities[0].maxFrameRate,
+                minResolution=aSnapshotCapabilities[0].resolution,
+                maxResolution=aSnapshotCapabilities[0].resolution,
+                quality=90,
+            )
+            await self.send_single_cmd(endpoint=endpoint, cmd=snpStreamAllocateCmd)
+        except InteractionModelError as e:
+            asserts.assert_equal(e.status, Status.Success, "Unexpected error returned")
+            pass
+
+    @async_test_body
+    async def test_TC_AVSM_2_10(self):
+        endpoint = self.get_endpoint(default=1)
+        cluster = Clusters.CameraAvStreamManagement
+        attr = Clusters.CameraAvStreamManagement.Attributes
+        commands = Clusters.CameraAvStreamManagement.Commands
+
+        self.step("precondition")
+        # Commission DUT - already done
+        await self._precondition_one_allocated_snp_stream()
+
+        self.step(1)
+        aFeatureMap = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attr.FeatureMap)
+        logger.info(f"Rx'd FeatureMap: {aFeatureMap}")
+        snpSupport = (aFeatureMap & cluster.Bitmaps.Feature.kSnapshot) > 0
+        asserts.assert_true(snpSupport, "Snapshot Feature is not supported.")
+
+        self.step(2)
+        aAllocatedSnapshotStreams = await self.read_single_attribute_check_success(
+            endpoint=endpoint, cluster=cluster, attribute=attr.AllocatedSnapshotStreams
+        )
+        logger.info(f"Rx'd AllocatedSnapshotStreams: {aAllocatedSnapshotStreams}")
+        asserts.assert_equal(len(aAllocatedSnapshotStreams), 1, "The number of allocated snapshot streams in the list is not 1.")
+        aStreamID = aAllocatedSnapshotStreams[0].snapshotStreamID
+
+        self.step(3)
```

Comment:
Shouldn't we also check that captureSnapshotResponse.data is not None, and that the image codec and resolution are coherent?

Reply:
Thank you for your comments. We fixed it. You are right.